pennylane/tape/tape.py

```diff
@@ -1052,10 +1053,10 @@ def _shape_shot_vector_multi_homogenous(mps, device):
         elif ret_type == qml.measurements.Sample:
             shape = []
             for shot_val in device.shot_vector:
+                shots = shot_val.shots
                 for _ in range(shot_val.copies):
-                    shots = shot_val.shots
                     if shots != 1:
-                        shape.append(tuple([shots, len(mps)]))
+                        shape.append((shots, len(mps)))
                     else:
                         shape.append((len(mps),))
```

Comment:
I guess one could even pull out the case distinction? Along the lines of
```py
shots = shot_val.shots
if shots != 1:
    shape.extend((shots, len(mps)) for _ in range(shot_val.copies))
else:
    shape.extend((len(mps),) for _ in range(shot_val.copies))
```

Reply:
Nice catch! I implemented this in commit c233bc7ba5df9b406fcd1ea901d30197d49a3d01.